### ISSUE TITLE
[Help]:  large time processing to plot unwrapped phase  

### ISSUE THREAD

**Issue by mateosanchezospina** [mobigroup/gmtsar#151]
**Describe the problem you encountered**
"bas.plot_phases(unwrap_sbas.phase[:8], caption='SBAS Phase, [rad]')" , more than 45 mintues and still loading.

**OS and software version**
Please specify your operating system and the version of PyGMTSAR you are using, if applicable.
- OS:
- PyGMTSAR version: 

**Steps to reproduce the problem**
2D Unwrapping - sbas phase 
sbas.plot_phases(unwrap_sbas.phase[:8], caption='SBAS Phase, [rad]')

**Note**
- If you have a general question not directly related to PyGMTSAR, consider using the discussions page.

[For additional support, consider using the ChatGPT4-based InSAR AI Assistant at https://insar.dev/ai. It can provide guidance and help with understanding PyGMTSAR's capabilities and troubleshooting issues.]

**Comment by AlexeyPechnikov** [comment]
In the case of large rasters, SNAPHU unwrapping can require a lot of time, depending on your hardware. Have you materialized the dataset unwrap_sbas before plotting? While this does not speed up the process, it provides a progress indicator to help estimate the remaining time.

**Comment by mateosanchezospina** [comment]
Alexey thankyou very much. Probably is it, I didn´t the last step, materialized the dataset. I´ll try again.

**Comment by AlexeyPechnikov** [comment]
It seems you have large rasters because the first 8 cannot be completed in 45 minutes. What are the dimensions of your interferogram stack? Typically, SBAS resolution is 60 meters, and you do not need to unwrap overly large interferograms. Of course, you can use a more precise resolution, but it can be slow and is often not necessary since we can enhance lower-resolution SBAS results with higher-resolution PSI.

**Comment by mateosanchezospina** [comment]
I dont think so. The resolution rasters that I used is by default 60 m, and the stack numbers of interferoframs is 27. So probably shouldn't be delayed. I think the reason is that I did not execute the las step "materialized the dataset unwrap_sbas", because there was an "optional" notice in the text. So I´ll try again, in few momements, and I update the issue thread. Again, thank you for your help.

**Comment by mateosanchezospina** [comment]
Hello again. After trying several times, the line "unwrap_sbas = sbas.sync_cube(unwrap_sbas, 'unwrap_sbas')" still does not execute. Is loading for more than 30 minutes and the bar does not exceed 36%. I attach a screenshot, I hope there is some solution. Thank you. 
![screen](https://github.com/user-attachments/assets/3bb15dd3-84ff-49ed-89f0-7bedbe55fd12)

**Comment by AlexeyPechnikov** [comment]
Your data resolution is 15 meters (see your y and x spacing, 1 unit per y means about 15 meters and 4 units per x mean the same). As I’ve mentioned above, 60-meter resolution processing might be much faster.

**Comment by mateosanchezospina** [comment]
Ok I have it. But how can I change this resolution to 60 m? Update: I chose an AOI with a smaller dimension size, and it worked correctly. However, in LOS displacement decompose, there appears to be noise in the trends.
![SBAS_LOS_Dispkacement_Decompose_2022_2023](https://github.com/user-attachments/assets/779f6bd3-bc08-4323-a3a2-e3cf04fad5f4)
What you can appreciate?

**Comment by mateosanchezospina** [comment]
How could I read a document where you learn to interpret these results? (Notebook: Golden Valley)

**Comment by AlexeyPechnikov** [comment]
Check the ‘resolution’ parameter for the function compute_interferogram_multilook(). Ensure that your interferograms are clean from atmospheric noise, and use the function plot_baseline_displacement_los_mm() to check changes in single pixels.

I share some explanations of the results on Patreon: https://www.patreon.com/pechnikov

**Comment by Aragon-yggdrasill** [comment]
> 检查函数 compute_interferogram_multilook（） 的“resolution”参数。确保干涉图没有大气噪声，并使用函数 plot_baseline_displacement_los_mm（） 检查单个像素的变化。
> 
> 我在 Patreon 上分享了一些对结果的解释： https://www.patreon.com/pechnikov

![QQ截图20240723134808](https://github.com/user-attachments/assets/59042e0e-40c3-4a6a-807c-46791f5d48fe)
![QQ截图20240723134821](https://github.com/user-attachments/assets/e84627bd-f446-4929-bf6d-b96886519291)
I also tried adjusting other parameters, but either there was no change or they were all zero. The example used was LakeSarez_Landslides_2017.ipynb. @AlexeyPechnikov

**Comment by AlexeyPechnikov** [comment]
Your variable ‘landmask_sbas’ is defined on a different grid and needs to be adjusted here.

**Comment by Aragon-yggdrasill** [comment]
Thank you for your reply, the problem has been solved
